### PR TITLE
MDS 467 Generate Permit numbers for new NoW application permits

### PR DESCRIPTION
**Pull request by JustinMacaulay**
This adds a new sequence to the DB to keep track of permit numbers starting at 2000000. Also adds a section to the permit post resource to create a permit no if one was not provided and a now application guid was.

### CODE REVIEW

File: migrations/sql/V2020.10.29.09.24__add_permit_number_sequence copy.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (exzizt) [inline migrations/sql/V2020.10.29.09.24__add_permit_number_sequence copy.sql:1]: The filename has a "copy" at the end. :P

File: services/core-api/app/api/mines/permits/permit/models/permit.py [hunk 2]
```diff
@@ -157,11 +160,16 @@ def find_by_now_application_guid(cls, _now_application_guid):
         return None
 
     @classmethod
-    def create(cls, mine, permit_no, permit_status_code, add_to_session=True):
+    def create(cls, mine, permit_no, permit_status_code, permit_prefix, add_to_session=True):
         permit = cls.find_by_permit_no(permit_no)
+        number_value = None
```
Comment (exzizt) [inline services/core-api/app/api/mines/permits/permit/models/permit.py:165]: Maybe a variable name like `next_permit_no_sequence` is more descriptive of the intention?

File: services/core-api/app/api/mines/permits/permit/models/permit.py [hunk 3]
```diff
@@ -157,11 +160,16 @@ def find_by_now_application_guid(cls, _now_application_guid):
...
         if not permit:
+            if not permit_no:
+                number_value = db.session.execute(cls.permit_no_seq)
+                permit_no = permit_prefix + str(number_value) 
+                current_app.logger.debug(permit_no)
```
Comment (exzizt) [inline services/core-api/app/api/mines/permits/permit/models/permit.py:170]: Is this debug statement still really useful?

File: services/core-api/app/api/mines/permits/permit/resources/permit.py [hunk 4]
```diff
@@ -57,6 +58,8 @@ class PermitListResource(Resource, UserMixin):
         help='Title of the lead inspector for this permit.')
     parser.add_argument(
         'regional_office', type=str, location='json', help='The regional office for this permit.')
+    parser.add_argument(
+        'permit_is_exploration', type=bool, location='json', help='Wether the permit is an exploration permit or not.')
```
Comment (exzizt) [inline services/core-api/app/api/mines/permits/permit/resources/permit.py:62]: Wether --> Whether

File: services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js [hunk 5]
```diff
@@ -222,7 +212,9 @@ export class NOWPermitGeneration extends Component {
 
   startDraftPermit = () => {
     this.setState({ isLoaded: false });
+    console.log(this.props.preDraftFormValues);
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:215]: Remove console logs in this file

**Review by exzizt — CHANGES_REQUESTED** [review]
Looks good! Just a few comments to address. :)

**Review by nayr974 — APPROVED** [review]
image

File: services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js (hunk 5 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:215]: Removed

File: services/core-api/app/api/mines/permits/permit/resources/permit.py (hunk 4 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/mines/permits/permit/resources/permit.py:62]: changed

File: services/core-api/app/api/mines/permits/permit/models/permit.py (hunk 3 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/mines/permits/permit/models/permit.py:170]: removed

File: services/core-api/app/api/mines/permits/permit/models/permit.py (hunk 2 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/mines/permits/permit/models/permit.py:165]: changed

File: migrations/sql/V2020.10.29.09.24__add_permit_number_sequence copy.sql (hunk 1 above)
Comment (JustinMacaulay) [reply migrations/sql/V2020.10.29.09.24__add_permit_number_sequence copy.sql:1]: renamed, I will have to close and reopen the PR though

**Review by exzizt — APPROVED** [review]
Good work, thanks for addressing my comments. ✔️

**Review by exzizt — DISMISSED** [review]

**Review by exzizt — APPROVED** [review]
💥

**Pull request merged**